Source/driver/Castro_sdc.cpp

```diff
@@ -0,0 +1,175 @@
+#include "Castro.H"
+#include "Castro_F.H"
+
+using namespace amrex;
+
+void
+Castro::do_sdc_update(int m_start, int m_end, Real dt_m) {
+
+  // this routine needs to do the update from time node m to m+1
+  //
+  // We come in with:
+  //   A_new[m_start] : this is the advective update at node m_start
+  //   A_old[:] : this is the advective source for all nodes at the old iterate
+  //
+  //   R_old[:] : this is the reaction source for all nodes at the old iterate
+
+  // If we do advection only, then the update is explicit.  If we do
+  // reactions, then the update is implicit within a zone.
+
+  // for 4th order reactive SDC, we need to first compute the source, C
+  // and do a ghost cell fill on it
+
+#ifdef REACTIONS
+  // SDC_Source_Type is only defined for 4th order
+  MultiFab tmp;
+  MultiFab& C_source = (fourth_order == 1) ? get_new_data(SDC_Source_Type) : tmp;
+
+  if (fourth_order == 1) {
+
+    // for 4th order reacting flow, we need to create the "source" C
+    // as averages and then convert it to cell centers.  The cell-center
+    // version needs to have 2 ghost cells
+    for (MFIter mfi(*k_new[0]); mfi.isValid(); ++mfi) {
+
+      const Box& bx = mfi.tilebox();
+
+      ca_sdc_compute_C4(BL_TO_FORTRAN_BOX(bx),
+                        BL_TO_FORTRAN_3D((*A_new[m_start])[mfi]),
+                        BL_TO_FORTRAN_3D((*A_old[0])[mfi]),
```

Comment:
One thing to consider here is that if we want to consider extending this to arbitrary order in the future, this method doesn't extend well. We could consider defining a single MultiFab that has SDC_NODES components. The cost is that we spend a little more memory since we can't alias the `m == 0` components.

Reply:
I have no plans to go beyond 4th order.  I don't even what to think about what the spatial stencils look like

Reply:
going to defer this for later